ocs_ci/ocs/fiojob.py

```diff
@@ -0,0 +1,228 @@
+        f"to write {pvc_size} Gi data on OCS backed volume"))
+    return write_timeout
+
+
+def wait_for_job_completion(namespace, timeout, error_msg):
+    """
+    This is a WORKAROUND of particular ocsci design choices: I just wait
+    for one pod in the namespace, and then ask for the pod again to get
+    it's name (but it would be much better to just wait for the job to
+    finish instead, then ask for a name of the successful pod and use it
+    to get logs ...)
+
+    Returns: name of Pod resource of the finished job
+    """
+    ocp_pod = ocp.OCP(kind="Pod", namespace=namespace)
+    try:
+        ocp_pod.wait_for_resource(
+            resource_count=1,
+            condition=constants.STATUS_COMPLETED,
+            timeout=timeout,
+            sleep=30)
+    except TimeoutExpiredError as ex:
+        # report some high level error as well
+        logger.error(error_msg)
+        # TODO: log both describe and the output from the fio pods, as DEBUG
+        ex.message = error_msg
+        raise(ex)
+
+    # indentify pod of the completed job
+    pod_data = ocp_pod.get()
+    # explicit list of assumptions, if these assumptions are not met, the
+    # code won't work and it either means that something went terrible
+    # wrong or that the code needs to be changed
+    assert pod_data['kind'] == "List"
+    pod_dict = pod_data['items'][0]
+    assert pod_dict['kind'] == "Pod"
+    pod_name = pod_dict['metadata']['name']
+    logger.info(f"Identified pod name of the finished Job: {pod_name}")
+    pod_name = pod_dict['metadata']['name']
+
+    return pod_name
+
+
+def write_data_via_fio(fio_job_file, write_timeout, pvc_size, target_percentage):
+    """
+    Write data via fio Job (specified in ``tf`` tmp file) to reach desired
+    utilization level, and keep this level for ``minimal_time`` seconds.
+    """
+    # unix timestamp before starting the job so that one can check status
+    # prior the fio job run
+    fio_job_start_ts = time.time()
+
+    # deploy the fio Job to the cluster
+    fio_job_file.create()
+
+    # high level description of the problem, reported in case of a job failure
+    # or timeout
+    error_msg = (
+        f"Job fio failed to write {pvc_size} Gi data on OCS backed "
+        f"volume in expected time {write_timeout} seconds."
+        " If the fio pod were still runing"
+        " (see 'last actual status was' in some previous log message),"
+        " this is caused either by"
+        " severe product performance regression"
+        " or by a misconfiguration of the clusterr, ping infra team.")
+    pod_name = wait_for_job_completion(
+        fio_job_file.project.namespace, write_timeout, error_msg)
+
+    ocp_pod = ocp.OCP(kind="Pod", namespace=fio_job_file.project.namespace)
+    fio_output = ocp_pod.exec_oc_cmd(
+        f"logs {pod_name}", out_yaml_format=False)
+
+    # parse fio output
+    fio_report = fio_to_dict(fio_output)
+
+    logger.info(fio_report)
```

Comment:
How long is the FIO pod logs output?
I feel this could lead to console log flooding.
How about capturing some highlights of the FIO logs output as logger.info and the full output as logger.debug?

Reply:
There is no change in the way how fio report is logged since this code was introduced in   62ae8197c1ef7a0fd05ec9aa7b76d8a952940e46.

The output is pushed as a dictionary dump into a single line and looks like this:

```
2020-04-07 18:54:02,380 - INFO - ocs_ci.ocs.fiojob.write_data_via_fio.192 - {'fio version': 'fio-3.7', 'timestamp': 1586278403, 'timestamp_ms': 1586278403976, 'time': 'Tue Apr  7 16:53:23 2020', 'jobs': [{'jobname': 'simple-write', 'groupid': 0, 'error': 0, 'eta': 0, 'elapsed': 142, 'job options': {'rw': 'write', 'buffered': '1', 'bs': '4k', 'ioengine': 'libaio', 'directory': '/mnt/target', 'fill_device': '1'}, 'read': {'io_bytes': 0, 'io_kbytes': 0, 'bw_bytes': 0, 'bw': 0, 'iops': 0.0, 'runtime': 0, 'total_ios': 0, 'short_ios': 0, 'drop_ios': 0, 'slat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0}, 'clat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0, 'percentile': {'1.000000': 0, '5.000000': 0, '10.000000': 0, '20.000000': 0, '30.000000': 0, '40.000000': 0, '50.000000': 0, '60.000000': 0, '70.000000': 0, '80.000000': 0, '90.000000': 0, '95.000000': 0, '99.000000': 0, '99.500000': 0, '99.900000': 0, '99.950000': 0, '99.990000': 0}}, 'lat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0}, 'bw_min': 0, 'bw_max': 0, 'bw_agg': 0.0, 'bw_mean': 0.0, 'bw_dev': 0.0, 'bw_samples': 0, 'iops_min': 0, 'iops_max': 0, 'iops_mean': 0.0, 'iops_stddev': 0.0, 'iops_samples': 0}, 'write': {'io_bytes': 6232281088, 'io_kbytes': 6086212, 'bw_bytes': 44087386, 'bw': 43054, 'iops': 10763.529096, 'runtime': 141362, 'total_ios': 1521554, 'short_ios': 0, 'drop_ios': 0, 'slat_ns': {'min': 3487, 'max': 41815110143, 'mean': 91325.245574, 'stddev': 42495148.45626}, 'clat_ns': {'min': 783, 'max': 3587294, 'mean': 971.448556, 'stddev': 6218.331569, 'percentile': {'1.000000': 812, '5.000000': 820, '10.000000': 828, '20.000000': 844, '30.000000': 844, '40.000000': 860, '50.000000': 860, '60.000000': 876, '70.000000': 892, '80.000000': 916, '90.000000': 1096, '95.000000': 1208, '99.000000': 1544, '99.500000': 2192, '99.900000': 9152, '99.950000': 14400, '99.990000': 105984}}, 'lat_ns': {'min': 4410, 'max': 41815116594, 'mean': 92431.957121, 'stddev': 42495174.939994}, 'bw_min': 360, 'bw_max': 709213, 'bw_agg': 100.0, 'bw_mean': 101094.7, 'bw_dev': 184718.726845, 'bw_samples': 120, 'iops_min': 90, 'iops_max': 177303, 'iops_mean': 25273.641667, 'iops_stddev': 46179.654526, 'iops_samples': 120}, 'trim': {'io_bytes': 0, 'io_kbytes': 0, 'bw_bytes': 0, 'bw': 0, 'iops': 0.0, 'runtime': 0, 'total_ios': 0, 'short_ios': 0, 'drop_ios': 0, 'slat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0}, 'clat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0, 'percentile': {'1.000000': 0, '5.000000': 0, '10.000000': 0, '20.000000': 0, '30.000000': 0, '40.000000': 0, '50.000000': 0, '60.000000': 0, '70.000000': 0, '80.000000': 0, '90.000000': 0, '95.000000': 0, '99.000000': 0, '99.500000': 0, '99.900000': 0, '99.950000': 0, '99.990000': 0}}, 'lat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0}, 'bw_min': 0, 'bw_max': 0, 'bw_agg': 0.0, 'bw_mean': 0.0, 'bw_dev': 0.0, 'bw_samples': 0, 'iops_min': 0, 'iops_max': 0, 'iops_mean': 0.0, 'iops_stddev': 0.0, 'iops_samples': 0}, 'sync': {'lat_ns': {'min': 0, 'max': 0, 'mean': 0.0, 'stddev': 0.0, 'percentile': {'1.000000': 0, '5.000000': 0, '10.000000': 0, '20.000000': 0, '30.000000': 0, '40.000000': 0, '50.000000': 0, '60.000000': 0, '70.000000': 0, '80.000000': 0, '90.000000': 0, '95.000000': 0, '99.000000': 0, '99.500000': 0, '99.900000': 0, '99.950000': 0, '99.990000': 0}}, 'total_ios': 0}, 'usr_cpu': 1.907881, 'sys_cpu': 5.628144, 'ctx': 6661, 'majf': 0, 'minf': 21, 'iodepth_level': {'1': 100.0, '2': 0.0, '4': 0.0, '8': 0.0, '16': 0.0, '32': 0.0, '>=64': 0.0}, 'latency_ns': {'2': 0.0, '4': 0.0, '10': 0.0, '20': 0.0, '50': 0.0, '100': 0.0, '250': 0.0, '500': 0.0, '750': 0.0, '1000': 88.161971}, 'latency_us': {'2': 11.247317, '4': 0.385856, '10': 0.120272, '20': 0.053695, '50': 0.013802, '100': 0.01, '250': 0.01, '500': 0.01, '750': 0.01, '1000': 0.01}, 'latency_ms': {'2': 0.01, '4': 0.01, '10': 0.0, '20': 0.0, '50': 0.0, '100': 0.0, '250': 0.0, '500': 0.0, '750': 0.0, '1000': 0.0, '2000': 0.0, '>=2000': 0.0}, 'latency_depth': 1, 'latency_target': 0, 'latency_percentile': 100.0, 'latency_window': 0}], 'disk_util': [{'name': 'rbd2', 'read_ios': 4, 'write_ios': 617, 'read_merges': 0, 'write_merges': 802, 'read_ticks': 41424, 'write_ticks': 17728394, 'in_queue': 17769501, 'util': 0.763688}]}
```

That said, we could log just overall disk utilization stats in info level like this:

```
2020-04-07 18:54:02,380 - INFO - ocs_ci.ocs.fiojob.write_data_via_fio.192 - fio disk_util stats: [{'util': 0.763688, 'read_ticks': 41424, 'read_merges': 0, 'name': 'rbd2', 'write_ios': 617, 'in_queue': 17769501, 'write_merges': 802, 'read_ios': 4, 'write_ticks': 17728394}]
```